go-controller/pkg/node/gateway_shared_intf_linux.go

```diff
@@ -133,29 +133,46 @@ func deleteLocalNodeAccessBridge() error {
 	return nil
 }
 
-func addSharedGatewayIptRules(service *kapi.Service) {
-	rules := getGatewayIPTRules(service, nil)
+// If A Service External Traffic Policy==Local and backend is a host-networked pod
+// we must steer traffic from host -> svc straight to the host instead of into OVN
+// via the cluster IP of the service
```

Comment:
nit: rm line 138 we use the iptables rules here to DNAT directly to Host networked pod